### PR TITLE
Dp195 260 add address page

### PR DESCRIPTION
**Pull request by moiseienko-dg**

### CODE REVIEW

File: oct/pages/models/add_address_page.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        "last_name": {"locator": ("ID", "input-lastname"), "class": InputField},
+        "company": {"locator": ("ID", "input-company"), "class": InputField},
+        "address_1": {"locator": ("ID", "input-address-1"), "class": InputField},
+        "address_2": {"locator": ("ID", "input-address-2"), "class": InputField},
+        "city": {"locator": ("ID", "input-city"), "class": InputField},
+        "post_code": {"locator": ("ID", "input-postcode"), "class": InputField},
+        "country": {"locator": ("ID", "input-country"), "class": DropDown},
+        "region_state": {"locator": ("ID", "input-zone"), "class": DropDown},
+        "default_address": {
+            "XPATH": ("ID", '//*[@id="content"]/form/fieldset/div[10]/div'),
```
Comment (zoiaivanova) [inline oct/pages/models/add_address_page.py]:
```suggestion
            "locator": ("XPATH", '//*[@id="content"]/form/fieldset/div[10]/div'),
```

File: oct/pages/models/add_address_page.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+
+class DefaultAddress(RadioButtonGroup):
+    contains = {
+        "yes": {
+            "locator": ("XPATH", '//*[@id="content"]/form/fieldset/div[10]/div/label[1]/input'),
+            "class": RadioButton,
+        },
+        "no": {
+            "locator": ("XPATH", '//*[@id="content"]/form/fieldset/div[10]/div/label[2]/input'),
```
Comment (zoiaivanova) [inline oct/pages/models/add_address_page.py]: Use a relative path, not an absolute one. This is almost the same as its parent element (https://github.com/ITA-Dnipro/dp-195-taqc-python/pull/13/files#diff-4c02beb94e979dbc036c36794765b46a6e6e120caf5ec651237ebbfd0d002e96R30)

File: oct/pages/models/add_address_page.py (hunk 1 above)
Comment (moiseienko-dg) [reply oct/pages/models/add_address_page.py]: fixed

File: oct/pages/base/elements.py [hunk 3]
```diff
@@ -100,3 +99,9 @@ def href(self):
         if link is None:
             link = self._base.get_attribute("value")
         return link
+
+
+class DropDown(Element):
+    def select(self, name):
+        """Click it"""
+        self._base.find_element_by_xpath(f'//*[@id="input-zone"]/option[text()="{name}"]').click()
```
Comment (zoiaivanova) [inline oct/pages/base/elements.py]: This selector leads only to "input-zone" dropdown but it should be reusable for every <select> tag
Снимок экрана 2021-02-03 в 02 31 51

Comment (zoiaivanova) [reply oct/pages/base/elements.py]: Снимок экрана 2021-02-03 в 02 31 51

File: oct/pages/models/add_address_page.py (hunk 2 above)
Comment (moiseienko-dg) [reply oct/pages/models/add_address_page.py]: fixed

File: oct/pages/base/elements.py (hunk 3 above)
Comment (moiseienko-dg) [reply oct/pages/base/elements.py]: your truth,  fixed

File: oct/pages/models/add_address_page.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
+from pages.base.page import BasePage
+from pages.base.elements import RadioButton, Form, InputField, RadioButtonGroup, DropDown
+
+
+class DefaultAddress(RadioButtonGroup):
+    contains = {
+        "yes": {
+            "locator": ("XPATH", "div/label[1]/input"),
```
Comment (zoiaivanova) [inline oct/pages/models/add_address_page.py]: Relative xpath starts with "//" (the same issue in line 12). And div seems to be redundant in both lines:
```suggestion
            "locator": ("XPATH", "//label[1]/input"),
```

Comment (moiseienko-dg) [reply oct/pages/models/add_address_page.py:8]: Unfortunately, github do not have "handface" smile :D

Comment (moiseienko-dg) [reply oct/pages/models/add_address_page.py]: Unfortunately, github do not have "handface" smile :D

It should be only one "/" in the path, no?: "/label[1]/input"

Comment (zoiaivanova) [reply oct/pages/models/add_address_page.py]: Probably "/" should work too. But xpath shouldn't start without "/" or "//" slashes.

> Single Slash “/” – Single slash is used to create Xpath with absolute path i.e. the xpath would be created to start selection from the document node/start node.
> Double Slash “//” – Double slash is used to create Xpath with relative path i.e. the xpath would be created to start selection from anywhere within the document.

Comment (moiseienko-dg) [reply oct/pages/models/add_address_page.py]: done

**Review by N1ki4 — APPROVED** [review]

**Review by YuriiShp — APPROVED** [review]

**Comment by YuriiShp** [comment]
Please, merge updated 'dev' branch to avoid conflicts

**Review by zoiaivanova — APPROVED** [review]

**Review by SerhiiKhyzhko — APPROVED** [review]

**Pull request merged**